packages/did-session/src/index.ts

```diff
@@ -109,6 +142,72 @@ export class DIDSession {
     return this.#did
   }
 
+  /**
+   * Serialize session into string, can store and initalize the same session again while valid
+   */
+  serialize(): string {
+    if (!this.#keySeed || !this.#cacao) throw new Error('No session to seralize')
+    const session = {
+      sessionKeySeed: bytesToBase64(this.#keySeed),
+      cacao: this.#cacao,
+    }
+    return JSONToBase64url(session)
+  }
+
+  /**
+   * Initialize a session from a serialized session string
+   */
+  static fromSession(session: string, authProvider: EthereumAuthProvider) {
+    const { sessionKeySeed, cacao } = base64urlToJSON(session) as SessionObj
+    return new DIDSession({ authProvider, cacao, keySeed: base64ToBytes(sessionKeySeed) })
+  }
+
+  /**
+   * Determine if a session is expired or not
+   */
+  get isExpired(): boolean {
+    if (!this.#cacao) throw new Error('No session available')
+    const expTime = this.#cacao.p.exp
+    if (!expTime) return false
+    return Date.parse(expTime) < Date.now()
+  }
+
+  /**
+   * Number of seconds until a session expires
+   */
+  get expireInSecs(): number {
```

Comment:
will be making exp required in cacao after, so will be better at that time

and yeah I guess it could also just return a Date instead, as suggested